cstool/cstool.c

```diff
@@ -114,130 +123,177 @@ int main(int argc, char **argv)
 	uint64_t address = 0;
 	cs_insn *insn;
 	cs_err err;
-	bool x86_arch = false;
+    cs_mode md;
+    char *arch;
+    bool debug_flag = false;
 
-	if (argc != 3 && argc != 4) {
+	if (argc != 3 && argc != 4 && argc != 5) {
 		usage(argv[0]);
 		return -1;
 	}
 
-	mode = argv[1];
-	assembly = preprocess(argv[2], &size);
-	if (assembly == NULL) {
-		printf("ERROR: invalid assembler-string argument, quit!\n");
-		return -3;
-	}
-
-	if (argc == 4) {
-		// cstool <arch> <assembly> <address>
-		char *temp;
-		address = strtoull(argv[3], &temp, 16);
-		if (temp == argv[3] || *temp != '\0' || errno == ERANGE) {
-			printf("ERROR: invalid address argument, quit!\n");
-			return -2;
-		}
-	}
-
+    if (!strcmp(argv[1], "-d")) {
+        if (argc == 3) {
+            usage(argv[0]);
+            return -1;
+        }
+        debug_flag = true;
+        mode = argv[2];
+        assembly = preprocess(argv[3], &size);
+        if (argc == 5) {
+            char *temp;
+            address = strtoull(argv[4], &temp, 16);
+            if (temp == argv[4] || *temp != '\0' || errno == ERANGE) {
+                printf("ERROR: invalid address argument, quit!\n");
+                return -2;
+            }
+        }
+    } else {
+        if (argc == 5) {
+            usage(argv[0]);
+            return -1;
+        }
+        
+        mode = argv[1];
+        assembly = preprocess(argv[2], &size);
+        if (assembly == NULL) {
+            printf("ERROR: invalid assembler-string argument, quit!\n");
+            return -3;
+        }
+        
+        if (argc == 4) {
+            // cstool <arch> <assembly> <address>
+            char *temp;
+            address = strtoull(argv[3], &temp, 16);
+            if (temp == argv[3] || *temp != '\0' || errno == ERANGE) {
+                printf("ERROR: invalid address argument, quit!\n");
+                return -2;
+            }
+        }
+    }
+	
 	if (!strcmp(mode, "arm")) {
-		err = cs_open(CS_ARCH_ARM, CS_MODE_ARM, &handle);
+        arch = "arm";
```

Comment:
looks like you are using space, rather than TAB, for indentation? the code is not aligned here. could you fix this issue (at many places)?